wireapp/wire-ios-mono
# fix: link preview is sent but marked as not sent - WPB-10495

<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-10495" title="WPB-10495" target="_blank"><img alt="Bug" src="https://wearezeta.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/10803?size=medium" />WPB-10495</a>  [iOS] Links marked as not sent, but were sent
  </td></table>
  <br />
 













### Issue

Given that you send a link preview and it is successfully sent, if you want for about 1 minute, then restart the app, the link preview is marked as not sent.

### Cause

When sending a message, an expiration date is set on the message. Normally when a message is successfully sent, it is marked as sent, which will remove the expiration timer.

In the case of link previews however, the message is (for some reason I don't understand right now) sent twice. First, it is sent like a text message in `ClientMessageRequestStrategy` and is correctly marked as sent upon completion. But because it is a link preview, the `LinkPreviewUpdateRequestStrategy` will send it again, which will add the expiration date once more, and upon success, it will only update the link preview state to `done` but will not remove the expiration date. When restarting the app after this expiration date is reached, then the message will be considered expired and therefore the error will be shown.

### Solution

After the `LinkPreviewUpdateRequestStrategy` sends the message, mark it as sent. This will not only remove the expiration date, but also set the link preview state to `done`.


### Testing

1. Ensure your user settings allows link preview generation
2. Send a message that generates a link preview with an image (eg www.apple.com)
3. Ensure the message and link preview are sent and received.
4. Wait about 1 or 2 minutes.
5. Restart the app.
6. Ensure the link preview is not marked as expired.

---

### Checklist


---

### UI accessibility checklist

_If your PR includes UI changes, please utilize this checklist:_
